src/main/java/com/iexec/worker/compute/DockerService.java

```diff
@@ -87,34 +87,38 @@ public boolean isImagePulled(String image) {
      * 
      * maxExecutionTime != 0: we wait for the execution to end or we stop 
      *      the container when it reaches the deadline.
+     * 
+     * @return
+     * - Valid Optional object (with an compute stdout or empty string) when success.
+     * - Optional.empty() if failure.
      */
 
-    public DockerExecutionResult execute(DockerExecutionConfig dockerExecutionConfig) {
-        String chainTaskId = dockerExecutionConfig.getChainTaskId();
-        String containerName = dockerExecutionConfig.getContainerName();
-        long maxExecutionTime = dockerExecutionConfig.getMaxExecutionTime();
+    public Optional<String> run(DockerCompute dockerCompute) {
```

Comment:
What you think about just a boolean return?
Stdout is stored on the disk, we could read it with taskId whenever we want instead of putting it in-memory (?)

Reply:
Good idea, but at this point, the stdout is not written on the disk yet.